src/main/scala/jio/package.scala

```diff
@@ -74,7 +64,37 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     }
   }
 
-  trait Pathish[Rep] {
+  trait Metadataish {
+    def exists: Boolean
```

Comment:
Yeah, I have been thinking about that. I would however make some constructs worse. Take for example:
```
... match {
  case f if f.exists => effect(eok)(f.delete())
  case _             => doesNotExist()}
```
This would become:
```
... match {
  case NoMetadata => doesNotExist()}
  case f             => effect(eok)(f.delete())
}
```

It would become even more problematic when `...` is a path which will never be `NoMetadata`, but could be the equivalent of `NoMetadata`.

Maybe I'm misunderstanding your intention, if so, let me know.